yt_dlp/extractor/zingmp3.py

```diff
@@ -1,22 +1,50 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import datetime
+import hashlib
+import hmac
+import re
+from urllib.parse import urlencode
+
 from .common import InfoExtractor
 from ..utils import (
-    int_or_none,
+    int_or_none, try_get,
 )
 
+SLUG_API = {
+    "bai_hat": "/api/v2/page/get/song",
+    "video_clip": "/api/v2/page/get/video",
+    "playlist": "/api/v2/page/get/playlist",
+    "lyric": "/api/v2/lyric/get/lyric",
+    "song_streaming": "/api/v2/song/get/streaming",
+}
+
+TIMESTAMP = str(int(datetime.datetime.now().timestamp()))
+API_KEY = '88265e23d4284f25963e6eedac8fbfa3'
+SECRET_KEY = b'2aa2d1c561e809b267f3638c4a307aab'
+
 
 class ZingMp3BaseIE(InfoExtractor):
-    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?:%s)/[^/]+/(?P<id>\w+)\.html'
+    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?P<type>(?:%s))/[^/]+/(?P<id>\w+)(?:\.html|\?)'
     _GEO_COUNTRIES = ['VN']
+    _IS_UPDATE_COOKIES = False
+    _DOMAIN = 'https://zingmp3.vn'
 
     def _extract_item(self, item, fatal):
-        item_id = item['id']
-        title = item.get('name') or item['title']
+        item_id = item['encodeId'] or self.song_id
```

Comment:
```suggestion
        item_id = item.get('encodeId') or song_id
```
Pass `song_id` as an argument. You must change all related references.

Reply:
Done.